src/cmd/cloud.js

```diff
@@ -779,11 +779,24 @@ module.exports = class CloudCommand extends CLICommandBase {
 	 */
 	_processDirIncludes(fileMapping, dirname, { followSymlinks } = {}){
 		dirname = path.resolve(dirname);
+		let files = new Set();
 
-		const includesFile = path.join(dirname, settings.dirIncludeFilename);
-		const ignoreFile = path.join(dirname, settings.dirExcludeFilename);
-		let hasIncludeFile = false;
+		this._getDefaultIncludes(files, dirname, { followSymlinks });
```

Comment:
Is the intention to stack the settings on top of each other?  For example, the `_getDefaultIncludes()` will add a bunch of patterns then `_getCustomIncludes()` will add to the existing pattern list, and so on?  Or if the custom list is present it will clear any of the default patterns?

Reply:
The custom list will not clear the list of files added by the default list. The difference is in files added by the user system (_getDefaultIncludes) vs. user (_getCustomIncludes) 

Here's what you can expect from this functionality:
1. `_getDefaultIncludes` will add the default files as provided in the list into a resultant array that will get sent to the cloud compiler. Default list doesn't look at user inputs. It checks the patterns from the system.
2. `_getCustomIncludes` will include more files as per the patterns in the `particle.include` file as specified by the user. Files added in step1 are not cleared.
3. `_getDefaultIgnores` will remove any files as specified by the system (currently only lib/helper/examples/)
4. `_getCustomIgnores` will remove any files as specified by the user in the `particle.ignore` file.